Library/Homebrew/extend/os/linux/keg_relocate.rb

```diff
@@ -84,9 +86,9 @@ def self.relocation_formulae
   def self.bottle_dependencies
     @bottle_dependencies ||= begin
       formulae = relocation_formulae
-      gcc = Formula["gcc"]
+      gcc = Formula[preferred_gcc]
       if !Homebrew::EnvConfig.force_homebrew_on_linux? &&
-         DevelopmentTools.non_apple_gcc_version("gcc") < gcc.version.to_i
+         DevelopmentTools.non_apple_gcc_version(preferred_gcc) < gcc.version.to_i
```

Comment:
```suggestion
         DevelopmentTools.non_apple_gcc_version("gcc") < gcc.version.to_i
```
This one needs to stay as is, it tests the version of whichever `gcc --version` is found first in the `PATH`. Line 87 immediately above however needs to change from `gcc = Formula["gcc"]` to `gcc = Formulary.factory("preferred_gcc")`